### ISSUE TITLE
Support `run`ing multiple GATT services

### ISSUE THREAD

**Issue by dexterduck** [embassy-rs/nrf-softdevice#73]
Right now, you can register multiple GATT services and call their `set`/`get`/`notify` methods in parallel. However, you can't have multiple registered services subscribed to events since `gatt_server::run` uses a portal internally. 
A workaround right now is to create dummy wrapper server that dispatches to each individual service, but that's pretty hack-y.

It would be nice to have a run macro that could support multiple services, e.g.:
```rust
let srv_a: ServiceA = unwrap!(gatt_server::register(sd));
let srv_b: ServiceB = unwrap!(gatt_server::register(sd));
...
let res = gatt_server::run!(&conn,
  &srv_a, |_e| defmt::info!("Handle service A event"),
  &srv_b, |_e| defmt::info!("Handle service B event"),
).await;
```

As far as I can tell, this interface would be feasible to implement as the macro would just need to internally chain the servers' `on_write` methods.
